zbus_macros/src/iface.rs

```diff
@@ -689,40 +689,42 @@ pub fn expand(args: Punctuated<Meta, Token![,]>, mut input: ItemImpl) -> syn::Re
                         quote!(self.#ident()#method_await)
                     };
 
-                    let prop_changed_method = quote!(
-                        pub async fn #prop_changed_method_name(
-                            &self,
-                            signal_emitter: &#zbus::object_server::SignalEmitter<'_>,
-                        ) -> #zbus::Result<()> {
-                            let mut changed = ::std::collections::HashMap::new();
-                            let value = <#zbus::zvariant::Value as ::std::convert::From<_>>::from(#prop_value_handled);
-                            changed.insert(#member_name, value);
-                            #zbus::fdo::Properties::properties_changed(
-                                signal_emitter,
-                                #zbus::names::InterfaceName::from_static_str_unchecked(#iface_name),
-                                changed,
-                                ::std::borrow::Cow::Borrowed(&[]),
-                            ).await
-                        }
-                    );
+                    if p.emits_changed_signal != PropertyEmitsChangedSignal::Const {
```

Comment:
Also on `PropertyEmitsChangedSignal::False`, no? :thinking:

Reply:
I assumed false, means the signal doesn't gets emitted automatically but you still want the helpers to be generated, no?

Reply:
The spec says:

> If set to false, the org.freedesktop.DBus.Properties.PropertiesChanged signal, .. is not guaranteed to be emitted if the property changes.

which makes it vague. I can't imagine a use case for someone explicitly setting this attribute to `false` and then wanting to emit signals so I'd just suggest we don't generate when set to `false`.

Reply:
> If set to false, the org.freedesktop.DBus.Properties.PropertiesChanged signal, .. is not guaranteed to be emitted if the property changes.

The spec seems to correspond to my understanding that I explained above. 
- `const` => never changes, so no need for any signals.
- `true` => changes -> generate changed signal
- `invalidates` => changes -> generate invalidate signal
- `false` => doesn't change, but the value might be computed at a later time, so generate the signal so the user can emit it still.

But then, I don't really see the difference between `true` & `false`.

Reply:
> false => doesn't change, but the value might be computed at a later time, so generate the signal so the user can emit it still.

I don't agree with this interpretation since IMO the "doesn't change" part contradicts the rest. Also as I wrote above, I don't see the use case.

Having said that, if I don't really have a very strong opinion here. Maybe there is a use case that I don't see. :shrug: So if you prefer, we can skip the generation for both `const` and `invalidates` but not for `false`.

Reply:
I personally only care about const here, so I will follow whatever you think is correct.

Reply:
The problem i see with the commit I just pushed, is there is no way to have both `invalidate` & `changed` signals generated. But then I might be missing something in regards of the spec.

Reply:
> is there is no way to have both `invalidate` & `changed` signals generated

That's actually on purpose. If you declare `invalidates` for your property, the client will not expect to receive values of the property in the signal, so there is really no need for then doing what you declared you won't be doing. :)